src/ganon-build/CommandLineParser.cpp

```diff
@@ -0,0 +1,76 @@
+#include "CommandLineParser.hpp"
+
+#include <defaults/defaults.hpp>
+
+#include <cxxopts.hpp>
+
+std::optional< Config > CommandLineParser::parse( int argc, char** argv )
+{
+    cxxopts::Options options( "ganon-build", "Ganon builder" );
+
+    // clang-format off
+    options.add_options()
+        ( "e,seqid-bin-file", "Tab-separated file linking sequences and bin identifiers. The file should contain the following fields: Seq. Identifier <tab> Pos. Seq. Start <tab> Pos. Seq. End <tab> Bin Id", cxxopts::value< std::string >() )
+        ( "o,output-filter-file", "Output filter file", cxxopts::value< std::string >() )
+        ( "u,update-filter-file", "Previously generated filter file to be updated", cxxopts::value< std::string >()->default_value( "" ) )
+        ( "c,update-complete", "Old and new sequences are provided for updated bins (used to remove sequences)", cxxopts::value< bool >()->default_value( "false" ) )
+        ( "s,filter-size", "Final filter size in Megabytes (MB) [mutually exclusive --filter-size-bits]", cxxopts::value< uint64_t >()->default_value( "16" ) )
+        ( "b,filter-size-bits", "Final filter size in Bits (bit) [mutually exclusive --filter-size]", cxxopts::value< uint64_t >()->default_value( "0" ) )
+        ( "k,kmer-size", "k size", cxxopts::value< uint16_t >()->default_value( "19" ) )
+        ( "n,hash-functions", "Number of hash functions", cxxopts::value< uint16_t >()->default_value( "3" ) )
+        ( "t,threads", "Number of threads", cxxopts::value< uint16_t >()->default_value( "1" ) )
+        ( "verbose", "Verbose output mode", cxxopts::value<bool>()->default_value("false"))
+        ( "h,help", "Show help commands" )
+        ( "v,version", "Show current version" )
+        ( "reference-files", "reference-files", cxxopts::value< std::vector< std::string > >() );
+    // clang-format on
+
+    options.parse_positional( { "reference-files" } );
+    options.positional_help( "ref.fna[.gz] [ref2.fna[.gz] ... refN.fna[.gz]]" );
+
+    const auto args = options.parse( argc, argv );
+
+    if ( args.count( "help" ) || argc == 1 )
```

Comment:
For some reason, cxxopts sets argc to 1 after parse, so the tool does not run and always shows the help info.

Reply:
Meaning we have to store the original value of argc to check its contents after the parse command